manager/apis/app/v1beta1/fybrikapplication_types.go

```diff
@@ -122,7 +112,7 @@ type DatasetDetails struct {
 
 	// Reference to a secret where the credentials are stored
 	// +optional
-	SecretRef SecretRef `json:"secretRef,omitempty"`
+	SecretRef taxonomy.SecretRef `json:"secretRef,omitempty"`
```

Comment:
I will continue later, but here we break compatibility with original fybrikApplication status. 
The content of SecrerRef is the same, but packages are different.

Reply:
How do we break compatibility? I just moved a common structure outside the crd types.

Reply:
FybrikApplicationStatus contains map[string]DatasetDetails.
Originally, one of the DatasetDetails fields was `manager.apis.v1beta1.SecretRef`, now it is `pkg.model.taxonomy.SecretRef`. We can try to check it, but my guess that goLang will not like this substitution.